### DISCUSSION TITLE
I'm stuck on "Loading smart connections". Kindly requesting assistance.

### DISCUSSION THREAD

**Discussion by BrianMwangi21** [brianpetro/obsidian-smart-connections#768] [Q&A] [answered]
I updated to version 2.1.1 today and smart connections seems not to be working as expected. It says that the import is done but when open the smart connections panel, it just says "Loading smart connections" as shown in the screenshot below. 

![image](https://github.com/user-attachments/assets/0ddadbbf-9568-423b-af74-b1461daebd82)

I thought maybe the update went wrong so I uninstalled then installed it again but it still stays this way.

Someone please assist.

Note: I am running Linux Ubuntu.

**Comment by hittjwX** [comment]
Switch to Embedding Model `TaylorAI/bge-micro-v2` for both blocks and notes in settings. I've had this same problem with other models. You might need to give it time to get the Smart Chat to reset.

![image](https://github.com/user-attachments/assets/f001e7ba-6398-4db2-bc11-1ab2cdbda2f2)

**Reply by BrianMwangi21** [reply]
I've tried this and nothing has changed over the last 3 days. It's still stuck on "Loading Smart Connections"

The last pop up I see is of importing done. I don't see any notification of embedding happening even when I click "Refresh Embeddings"

**Reply by hittjwX** [reply, marked as answer]
Not good. While I had "Loading Smart Connections" hang a few times a few upgrades ago, it cleared with the configuration change. 

I'm troubleshooting after this point. Try ...

1. Close Obsidian and back up your vault.
2. Make sure you have your OpenRouter API. It's located in `.smart-connections/smart_env.json`, and OpenRouter will let you regenerate it.
3. Open Obsidian, disable and uninstall the Smart Connections plugin, close Obsidian.
4. Delete `.smart-connections` and `.smart-env`
5. Go to `.obsidian/plugins` to remove `smart-connections`
6. Restart Obsidian and ensure you don't see any Smart Connections.
7. Let Obsidian index everything before reinstalling Smart Connections from community plugins.

This should give you a fresh install from the latest software. With a base configuration, it shouldn't hang. I purged the `.smart-connections` folder before (because it had stale entries.) It takes a long time to rebuild.

I ignore `.smart-connections` when syncing my Obsidian on github and only have Smart Connections on one desktop machine with a GPU.  My laptop and writing workstation have minimal plugins. I also use "Meta: Llama 3.1 8B Instruct (free)" on my OpenRouter.

Before you refresh everything, please ensure you aren't using a paid model on OpenRouter. The rebuild will cost a bit. As I said, this is troubleshooting in the form of a hard reset. My hung "Loading Smart Connections" cleared changing the embeds.

Good luck.

**Reply by Dnarts** [reply]
Okay, I officially solved this problem by following your directions, but it not exactly helped, so I refreshed embeddings using the button in settings, and it did not function again until I switched to old transformers.
So, the original path is:


1. Close Obsidian and backup your vault.
2. Ensure you have the OpenRouter API. It is contained in.smart-connections/smart_env.json, and OpenRouter allows you to regenerate it.
3. Launch Obsidian, disable and uninstall the Smart Connections plugin, then shut it.
4. Delete smart-connections and smart-env.
5. Go to obsidian/plugins to remove smart connections.
6. Restart Obsidian and make sure there are no Smart Connections.
7. Let Obsidian index everything before reinstalling Smart Connections via community plugins.
8. Select "Force_Refresh" at the bottom of the settings.
9. Switched to "Legacy Transformers" (no GPU) 

UPDATE: Perhaps you should try n.9 as a step 1 and restart Obsidian; it might do the trick.

Settings:
![image](https://github.com/user-attachments/assets/af452d62-b4a9-4b62-b7fb-ac8112ce5dcf)

Im back in the game after month for this plugin being off for me

**Reply by BrianMwangi21** [reply]
Thanks @jwhco it worked like magic

**Reply by hittjwX** [reply]
I'm glad to hear it. This is the kind of plug-in that causes a bit of panic if it doesn't work after a while.

**Reply by Dnarts** [reply]
> I'm glad to hear it. This is the kind of plug-in that causes a bit of panic if it doesn't work after a while.

Dam, this working only without GPU (on v2 transformers)... =(

**Reply by brianpetro** [reply]
@Dnarts I would expect it to work with more devices in future versions 🌴

**Reply by therealsun** [reply]
Hi, 

Thx for the great plugin. Unfortunately I'm stuck right after enabling it (MacBook Air M1).  Can't even get to the plugin settings and therefore can't change any parameter. Tried in different vault, delete and reinstall, disable any other plugins and try all those previous things again, tried to directly change some settings in smart_env.json but those are resetting on every start. Also I don't have any `.smart-connections/` folder in vault root.

Can you please help ?

**Reply by brianpetro** [reply]
@therealsun look for the `.smart-env` folder, delte that and restart 🌴

**Reply by therealsun** [reply]
Already tried this one too. Tried to modify the `smart_env.json` inside too. No luck that way

**Reply by therealsun** [reply]
hey brian,
wanted to let you know that the problem has been fixed with the latest update :)

**Comment by christianngarcia** [comment]
Thanks for posting a solution. 😊

**Reply by BavariaR** [reply]
I have the same problem and working only without GPU (on v2 transformers) is not really a solution, especially for the embedding. I think the embedding is something I don't want to send to the cloud, so I would like to have it local. But actually, without GPU, I can only reasonably run BGE micro version 2, which is really the least powerful embedding model. Otherwise, all the others are taking way too long on startup. I have an NVIDIA graphics card, which has a GPU that is reasonably powerful and would be a good fit to accelerate this embedding. But I don't really understand why it's not possible to use the Legacy Transformer version 3, including GPU. What is the problem with it? How can we solve it? This discussion is already from 2024, so it's like one year old. Is there any progress?
